app/controllers/posts_controller.rb

```diff
@@ -20,7 +20,15 @@ def create
   private
 
   def timeline_posts
-    @timeline_posts ||= Post.all.ordered_by_most_recent.includes(:user)
+    friends_posts = []
+    user_posts = current_user.posts
+    friends = current_user.friends
+    friends.each do |friend|
+      friends_posts += friend.posts
+    end
+    posts = user_posts + friends_posts
+
+    @timeline_posts = posts
   end
```

Comment:
You can simplify that code.
Kindly add this into the `User` model
```rb
def friends_and_own_posts
  Post.where(user: (self.friends + self))
  # This will produce SQL query with IN. Something like: select * from posts where user_id IN (1,45,874,43);
end
```
and call it with ```current_user.friends_and_own_posts```

Reply:
![Screenshot from 2020-08-29 01-26-38](https://user-images.githubusercontent.com/51195150/91622314-cf451a00-e996-11ea-889d-de262336f516.png)
Thanks for the tips but i can't implement the solution as presented and explained in my code, i will leave the original implementation.